lib/open_project/configuration.rb

```diff
@@ -458,128 +175,54 @@ def reload_smtp_settings!
-      end
-
-      def load_config_from_file(filename, env, config)
-        if File.file?(filename)
-          file_config = YAML::load(ERB.new(File.read(filename)).result)
-          if file_config.is_a? Hash
-            config.deep_merge!(load_env_from_config(file_config, env))
-          else
-            warn "#{filename} is not a valid OpenProject configuration file, ignoring."
-          end
-        end
-      end
-
-      def load_env_from_config(config, env)
-        merged_config = {}
-
-        if config['default']
-          merged_config.deep_merge!(config['default'])
-        end
-        if config[env]
-          merged_config.deep_merge!(config[env])
-        end
-        merged_config
-      end
-
-      # Convert old mail settings
-      #
-      # SMTP Example:
-      # mail_delivery.smtp_settings.<key> is converted to smtp_<key>
-      # options:
-      # disable_deprecation_message - used by testing
-      def convert_old_email_settings(config, options = {})
-        if config['email_delivery']
-          unless options[:disable_deprecation_message]
-            ActiveSupport::Deprecation.warn 'Deprecated mail delivery settings used. Please ' +
-                                            'update them in config/configuration.yml or use ' +
-                                            'environment variables. See doc/CONFIGURATION.md for ' +
-                                            'more information.'
-          end
-
-          config['email_delivery_method'] = config['email_delivery']['delivery_method'] || :smtp
-
-          ['sendmail', 'smtp'].each do |settings_type|
-            settings_key = "#{settings_type}_settings"
-            if config['email_delivery'][settings_key]
-              config['email_delivery'][settings_key].each do |key, value|
-                config["#{settings_type}_#{key}"] = value
-              end
-            end
-          end
-          config.delete('email_delivery')
-        end
-      end
-
-      def cache_parameters(config)
+      def cache_parameters
         mapping = {
           'cache_expires_in_seconds' => %i[expires_in to_i],
           'cache_namespace' => %i[namespace to_s]
         }
         parameters = {}
         mapping.each_pair do |from, to|
-          if config[from]
+          if self[from]
             to_key, method = to
-            parameters[to_key] = config[from].method(method).call
+            parameters[to_key] = self[from].method(method).call
           end
         end
         parameters
       end
 
-      # Filters a hash with String keys by a key prefix and removes the prefix from the keys
-      def filter_hash_by_key_prefix(hash, prefix)
-        filtered_hash = {}
-        hash.each do |key, value|
-          if key.start_with? prefix
-            filtered_hash[key[prefix.length..-1]] = value
-          end
+      def method_missing(name, *args, &block)
+        setting_name = name.to_s.sub(/(=|\?)$/, '')
```

Comment:
It should probably raise a `NoMethodError` if name ends with a `=`.

Currently when trying to assign it fails with `ArgumentError: wrong number of arguments (given 1, expected 0)` because the getter gets called.

Reply:
same in `#respond_to_missing?`

Reply:
👍 done in [25315d9](https://github.com/opf/openproject/pull/10296/commits/25315d97c0c719bbce6d0a8486130015d8f6ce6f)